src/Azure.Functions.Cli/Actions/HostActions/StartHostAction.cs

```diff
@@ -480,31 +501,128 @@ public override async Task RunAsync()
             await runTask;
         }
 
+        private async Task<bool> ShouldStartChildProcessFromHostRuntime(bool isVerbose)
+        {
+            if (string.Equals(HostRuntime, "default", StringComparison.OrdinalIgnoreCase))
+            {
+                if (isVerbose)
+                {
+                    PrintVerboseForHostSelection(isVerbose, "out-of-process");
+                }
+                return false;
+            }
+            else if (string.Equals(HostRuntime, DotnetConstants.InProc8HostRuntime, StringComparison.OrdinalIgnoreCase))
+            {
+                if (await ShouldNet8InProcBeLaunched())
+                {
+                    if (isVerbose)
+                    {
+                        PrintVerboseForHostSelection(isVerbose, DotnetConstants.InProc8HostRuntime);
+                    }
+                    return true;
+                }
+                else
+                {
+                    throw new CliException($"Invalid config for running {DotnetConstants.InProc8HostRuntime} host.");
+                }
+            }
+            else if (string.Equals(HostRuntime, DotnetConstants.InProc6HostRuntime, StringComparison.OrdinalIgnoreCase))
+            {
+                if (isVerbose)
+                {
+                    PrintVerboseForHostSelection(isVerbose, DotnetConstants.InProc6HostRuntime);
+                }
+                return true;
+            }
+            else
+            {
+                throw new CliException($"Invalid host runtime '{HostRuntime}'. Valid values are 'default', '{DotnetConstants.InProc8HostRuntime}', '{DotnetConstants.InProc6HostRuntime}'.");
+            }
+        }
+
+        private void PrintVerboseForHostSelection(bool isVerbose, string hostRuntime)
+        {
+            if (isVerbose)
+            {
+                ColoredConsole.WriteLine(VerboseColor($"Selected {hostRuntime} host"));
+            }
+        }
+
+        private async Task<bool> ShouldNet8InProcBeLaunched()
+        {
+            // Start .NET 8 child process if InProc8 is enabled
+            if (ShouldLaunchInProcNet8AsChildProcess() && await IsInProcNet8Enabled())
+            {
+                return true;
+            }
+            return false;
+        }
+
+        private bool ShouldLaunchChildProcessAfterInferringHostRuntimeAsync(bool shouldLaunchNet8, bool isVerbose)
+        {
+            // We should try to infer if we run inproc6 host, inproc8 host, or OOP host (default)
+            var isOutOfProc = GlobalCoreToolsSettings.CurrentWorkerRuntime == WorkerRuntime.dotnet ? false : true;
+
+            // Check if the app is in-proc
+            if (!isOutOfProc)
```

Comment:
See my other comment about doing this check earlier.

Reply:
defined isOutOfProc earlier (still kept it in method for the logging though)